### ISSUE TITLE
The "focused node" when determining event target needs to be defined

### ISSUE THREAD

**Issue by foolip** [w3c/clipboard-apis#70]
https://w3c.github.io/clipboard-apis/#fire-a-clipboard-event says "Set target to the focused node, or the body element if no node has focus" but "focused node" isn't defined. Maybe it's [document.activeElement](https://html.spec.whatwg.org/multipage/interaction.html#dom-document-activeelement)?

**Comment by MatsPalmgren** [comment]
No, it should **not** be `document.activeElement`.  It should be `Selection.focusNode`. The reason is that you can have an activeElement without a selection, and these commands (Copy etc) _operates on the selection_ by [definition](https://w3c.github.io/clipboard-apis/#clipboard-event-copy).  [Here's a simple testcase](https://bug1541572.bmoattachments.org/attachment.cgi?id=9055632) to demonstrate this.  Firefox and Chrome both prints:

```
Current selection is 
Selection { anchorNode: null, anchorOffset: 0, focusNode: null, focusOffset: 0, isCollapsed: true, rangeCount: 0, type: "None", caretBidiLevel: 0 }
Current activeElement is 
<div id="playground1" class="playground" tabindex="0">
Got copy event, target is 
<body>
```

That said, `document.activeElement` and `Selection.focusNode` are usually the same node, but there are cases when they are not, e.g. when the Selection has been modified by JS or when activeElement was set by a user gesture and CSS `user-select:none` prevented it from being selected (as intended).

**Comment by rniwa** [comment]
@whsieh

**Comment by julienw** [comment]
I agree with what @MatsPalmgren says above, except it should likely be `Selection.anchorNode`, which represents the _start_ of the selection.

I agree that the target should always be `anchorNode`, no matter what the context is, because it makes it much more intuitive to work with this event.

For example you can look at [this widely use implementation of copy](https://github.com/sudodoki/copy-to-clipboard/blob/master/index.js): the focus is never set on the element, this means that we don't control where `copy` event would be sent. Setting the focus on the added element seems to be an extra work that would be otherwise unneeded. Another way would be to make the node editable, which seems also an unneeded work.
Instead the current behavior in Firefox and Chrome is that we send the `copy` event to the selection, [which makes it possible to handle it here](https://github.com/sudodoki/copy-to-clipboard/pull/76/files).

Also because both Chrome and Firefox implement it this way now, and changing it might break existing websites and apps. It would most certainly break ours and the linked change above.

**Comment by Kaiido** [comment]
Just to point out that it seems currently Safari, Chrome, and Firefox use the `Range.startContainer` and not the `anchorNode` nor the `focusNode` (which would differ based on the direction of the selection).
